mlflow/utils/environment.py

```diff
@@ -500,6 +501,29 @@ def _process_pip_requirements(
     return conda_env, pip_reqs, constraints
 
 
+def _find_duplicate_requirements(requirements):
+    """
+    Checks if duplicate base package requirements are specified in any list of requirements
+    and returns the list of duplicate base package names.
+    Note that git urls and paths to local files are not being considered for duplication checking.
+    """
+    base_package_names = []
+
+    for package in requirements:
+        if package.startswith("-") or package.startswith("/") or "://" in package:
+            # Skip flags, local file paths, and urls
```

Comment:
Do we need this? Seems like try / catch requirement parsing should work :)

Reply:
wow. yeah. 
Also, the private repo declarations actually get captured by the Requirement library. That's one additional dupe check! Thanks for this comment :D